docs/answers.rst

```diff
@@ -0,0 +1,53 @@
+Answers
+=======
+
+Extraction
+----------
+
+``coala`` documentation is in ``rST`` files. ``docutils`` is used to parse rst
+files.
+
+``answers.extraction.Extractor`` is the ``NodeVisitor`` that visits all the
+``section`` nodes. All the code text(``docutils.nodes.literal_block``) are
+extracted and stored separately. All the normal text(all nodes excluding
+``literal_block``) are extracted and stored separately. Filename, and section id
+is appended at the end of the text for backtracking the file from the text.
+
+.. note::
+   If the node has a nested section node, than the current node
+   is ignored because the nested section nodes will be visited later as well.
+
+Forming Graph
+-------------
+
+A huge graph is created from all the text that was extracted using the
+extraction module.
+
+Creating nodes and Edges
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+For natural language processing ``spacy`` is used.
+
+Iterating over all the tokens:
+If `head<https://en.wikipedia.org/wiki/Head_(linguistics)>`_ of the token is
+the same as the token, then the token is ignored. If not, then
+`lemmatized<https://en.wikipedia.org/wiki/Lemmatisation>`_ form of token and
+head of token are created and an edge is formed between them.
+
+Holding information
+~~~~~~~~~~~~~~~~~~~
+
+Each node contains information about the sections, codes and texts they appear
+in, all of which are stored as attributes of the node.
+
+``node.sections`` is a list of section ids.
+``node.code`` is a list of code of sections particular token appears in.
+``node.text`` is a list of text of sections in which the token appears.
+
+
+Scoring answers
+~~~~~~~~~~~~~~~
+
+A graph of question is formed the same way the graph of documentation is
+formed. Then one iterates over question graph's edges, and increments the score
```

Comment:
Okay this is not clear at all.
1. Iterate over graph edges to get pairs of nodes.
2. Search for shortest path in the documentation graph for those pair of nodes.
3. Consider all nodes in the shortest path and the unique location where they come from.
4. Increment the count of these locations.
5. Return the location with the largest count/most relevancy. 

Note: Pleases add how count relates to relevancy explanation as well.